src/check.h.in

```diff
@@ -669,6 +671,263 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
 
+/* Number of digits to output via printf */
+#ifndef CK_FLOATING_DIG
+# define CK_FLOATING_DIG 6
+#endif /* CK_FLOATING_DIG */
+
+/* Floating point number comparison macros with improved output
+ * compared to ck_assert(). */
+/* OP may be any comparison operator. */
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %.*Lg, %s == %.*Lg", \
+  #X" "#OP" "#Y, \
+  #X, (int)CK_FLOATING_DIG, _ck_x, \
+  #Y, (int)CK_FLOATING_DIG, _ck_y); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X==Y
+ *
+ * If not X==Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq(X, Y) _ck_assert_floating(X, ==, Y)
+/**
+ * Check two floating point numbers to determine if X!=Y
+ *
+ * If not X!=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ne(X, Y) _ck_assert_floating(X, !=, Y)
+/**
+ * Check two floating point numbers to determine if X<Y
+ *
+ * If not X<Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_lt(X, Y) _ck_assert_floating(X, <, Y)
+/**
+ * Check two floating point numbers to determine if X<=Y
+ *
+ * If not X<=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_le(X, Y) _ck_assert_floating(X, <=, Y)
+/**
+ * Check two floating point numbers to determine if X>Y
+ *
+ * If not X>Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_gt(X, Y) _ck_assert_floating(X, >, Y)
+/**
+ * Check two floating point numbers to determine if X>=Y
+ *
+ * If not X>=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ge(X, Y) _ck_assert_floating(X, >=, Y)
+
+/**
+ * Check precision P of floating point numbers for correctness.
+ *
+ * If not P >= 0 or not P <= |LDBL_MIN_10_EXP|, the test fails.
+ *
+ * @param P positive number of decimal places to be compared (precision)
```

Comment:
This could be simplified to say:
```
Checks that P, used to indicate the precision of a floating point number, is valid.
For P to be valid P>=0 and P<=|LDBL_MIN_10_EXP|
```

As this is not intended to be used directly by a user the ```@note``` and ```@since``` tags are unnecessary. As far as I am concerned this is an 'internal' API.